src/utl/include/utl/Logger.h

```diff
@@ -162,7 +125,9 @@ class Logger
     }
 
   template <typename... Args>
+#ifndef SWIGPYTHON
     __attribute__((noreturn))
+#endif
```

Comment:
I did try that but there are still templated methods that SWIG chokes on.

Reply:
The methods are still there, you are just removing the attribute.  What error do you get?

Reply:
Yes, I can get rid of the errors on those two functions, but then I get the same error on the first definition of metric method, the ever helpful "Syntax error in input". Shades of bison without proper error handling.

/home/macd/ext_synth/OpenROAD/src/utl/include/utl/Logger.h:182: Error: Syntax error in input(3).

I can also ignore this entire method and things seem to build OK and I can run the regressions. What is interesting is that SWIG does not choke on the other templated method definitions. I've got no clue how this might fail if we just skip this method in compiling the SWIG wrapper. Well OK, no link errors, so maybe it's OK?  What is unfortunate is that SWIG also defines SWIGPYTHON in the generated cxx file. I looked in vain for a define that would only ignore stuff during the SWIG preprocessing phase, which seems like a handy thing to have, but I found nothing.

So, if we really don't want to split out the ToolId enum into a separate header, I can put in these SWIG ifdefs. I'd be OK with that.

Reply:
Update: It builds and links OK, but falls over hard on the regression tests. Separate header for ToolId seems the way to go.

Reply:
I guess it has trouble with
```
            typename = std::enable_if_t<std::is_arithmetic<T>::value>>
```

maybe it expects an name even if we don't actually use it.